dojo_plugin/utils/awards.py

```diff
@@ -74,13 +60,26 @@ def get_belts():
     return result
 
 def update_awards(user):
-    current_belts = get_user_belts(user)
-    for belt in current_belts:
-        belt_award = Belts.query.filter_by(user=user, name=belt).first()
-        if belt_award:
+    current_belts = [belt.name for belt in Belts.query.filter_by(user=user)]
+    discord_user = get_discord_user(user.id)
+    discord_roles = get_discord_roles()
+    for belt, dojo_id in BELT_REQUIREMENTS.items():
+        belt_role = belt.title() + " Belt"
+        missing_role = discord_user and discord_roles.get(belt_role) not in discord_user["roles"]
```

Comment:
As mentioned, I think this fires for every belt role every time any user solves anything...

Reply:
Sure does, just like before.